### PR TITLE
ML-158 k-fold cross validation on datasets

### PR DESCRIPTION
**Pull request by enevoldsen1oo**

### CODE REVIEW

File: core/util/get_datasets.py [hunk 1]
```diff
@@ -29,6 +18,80 @@ def get_one_park_dataset(
...
+    x_train, y_train, x_val, y_val = [], [], [], []
+
+    # Loop over folds and append the training and validation sets in each split
+    for i in range(folds_num):
+        x_val_fold, y_val_fold = folds[i]
+        x_val.append(x_val_fold)
+        y_val.append(y_val_fold)
+
+        x_train_fold = [folds[j][0] for j in range(folds_num) if j != i]
+        y_train_fold = [folds[j][1] for j in range(folds_num) if j != i]
```
Comment (Tinggaard) [inline core/util/get_datasets.py:76]: nice

File: core/util/get_datasets.py [hunk 2]
```diff
@@ -29,6 +18,80 @@ def get_one_park_dataset(
...
+        x_val,
+        y_val,
+        x_test,
+        y_test,
+    )
+
+
+def get_one_cross_park(
+    x: np.ndarray, y: np.ndarray, folds_num: int
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
```
Comment (Tinggaard) [inline core/util/get_datasets.py:43]: Could the return type be changed to `torch.Tensor().float()`, such that we don't have to do that every time we call this function in a notebook?

**Review by Tinggaard — CHANGES_REQUESTED** [review]
xdddd

File: core/util/get_datasets.py (hunk 2 above)
Comment (Maxnm47) [reply core/util/get_datasets.py:43]: snip2

**Review by MikkelBaekgaardJoergensen — APPROVED** [review]

**Review by MikkelBaekgaardJoergensen — APPROVED** [review]

**Review by Tinggaard — APPROVED** [review]

**Pull request merged**